### DISCUSSION TITLE
[feature request] Customizable prefix bounds

### DISCUSSION THREAD

**Discussion by villarragut** [minad/tempel#126] [Ideas]
When editing LaTeX, it's usual to insert templates for fractions or integrals. Unsurprisingly, mine are called fraction and integral, respectively. E.g. the template for fractions is
`(fraction "\\frac{" p "}{" p "}")`

There is a situation when you have something like
`$1+fraction$`
and the cursor is between n and the second $. Then, you want to call `tempel-complete` (or `tempel-expand`) and get
`$1+\frac{}{}$`

However, it does not work, since fraction is not identified as the prefix.

A workaround is to replace the last line in the function `tempel--prefix-bounds` from this:
`(bounds-of-thing-at-point 'symbol)))`
to this:
`(bounds-of-thing-at-point 'word)))`

It'd be nice to have a variable to customize this behavior. If its default value were `'symbol`, then it wouldn't break any setups.

This is a wonderful package. Thanks for the hard work!

Best wishes,
Victor

**Comment by minad** [comment]
Thanks for the suggestion. I am not sure if adding a customization option is justified, also since we already have the option `tempel-trigger-prefix`. It may make sense to generalize `tempel-trigger-prefix` instead, renaming it to `tempel-prefix-regexp`. It could be a regexp like for example `<\(\w+\)` (word with prefix), `\(\w+\)` (word without prefix), `\(\s_+\)` (symbol without prefix).

For now you can use this:

~~~ elisp
(advice-add #'tempel--prefix-bounds :override (lambda () (bounds-of-thing-at-point 'word))
~~~

**Reply by villarragut** [reply]
Thank you for your prompt response!

Indeed, it looks like a more sensible approach. And thanks for the code snippet! I'm not too proficient in Elisp and it comes in very handy.
